internal/fs/fs.go

```diff
@@ -1675,7 +1675,8 @@ func (fs *fileSystem) createLocalFile(
 
 	fullName := inode.NewFileName(parent.Name(), name)
 	child, ok := fs.localFileInodes[fullName]
-	if !ok {
+	// Create a new inode if a local file is unlinked and then recreated with the same name.
+	if !ok || (child != nil && child.(*inode.FileInode).IsUnlinked()) {
```

Comment:
We can reduce nesting by returning quickly: https://testing.googleblog.com/2017/06/code-health-reduce-nesting-reduce.html

if !(!ok || (child != nil && !child.(*inode.FileInode).IsUnlinked())) {
  return
}

The predicate can probably be simplified further by applying the [De Morgan's law](https://en.wikipedia.org/wiki/De_Morgan%27s_laws)

Reply:
+1 Agree with short circuits part - you should always write shorter blocks first. 
but multiple negations in if clause are even complicated to understand.

child != nil condition is not required since we are already checking ok .
So you can simplify as 
if ok && !child.(*inode.FileInode).isUnlinked() {
  return
}